packages/terminal/src/browser/terminal-preferences.ts

```diff
@@ -17,9 +17,58 @@
 /* eslint-disable max-len */
 
 import { interfaces } from '@theia/core/shared/inversify';
-import { createPreferenceProxy, PreferenceProxy, PreferenceService, PreferenceContribution, PreferenceSchema } from '@theia/core/lib/browser';
+import { IJSONSchema } from '@theia/core/lib/common/json-schema';
+import { createPreferenceProxy, PreferenceProxy, PreferenceService, PreferenceContribution, PreferenceSchema, PreferenceSchemaProperties } from '@theia/core/lib/browser';
 import { nls } from '@theia/core/lib/common/nls';
 import { editorGeneratedPreferenceProperties } from '@theia/editor/lib/browser/editor-generated-preference-schema';
+import { OS } from '@theia/core';
+import { terminalAnsiColorMap } from './terminal-theme-service';
+
+const commonProfileProperties: PreferenceSchemaProperties = {
+    env: {
+        type: 'object',
+        additionalProperties: {
+            type: 'string'
+        },
+        markdownDescription: nls.localize('theia/terminal/profileEnv', 'An object with environment variables that will be added to the terminal profile process. Set to `null` to delete environment variables from the base environment.'),
+    },
+    overrideName: {
+        type: 'boolean',
+        description: nls.localizeByDefault('Controls whether or not the profile name overrides the auto detected one.')
+    },
+    icon: {
+        type: 'string',
+        markdownDescription: nls.localize('theia/terminal/profileIcon', 'A codicon ID to associate with the terminal icon.  \nterminal-tmux:"$(terminal-tmux)"')
```

Comment:
minor:

```suggestion
        markdownDescription: nls.localize('theia/terminal/profileIcon', 'A codicon ID to associate with the terminal icon. \nterminal-tmux:"$(terminal-tmux)"')
```

Reply:
What's the difference? Looks identical to me 🤷

Reply:
No problem, the extra whitespace.

Reply:
Ah, I see.